exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -252,13 +252,36 @@ func (prwe *PrwExporter) handleSummaryMetric(tsMap map[string]*prompb.TimeSeries
 }
 
 // export sends a Snappy-compressed WriteRequest containing TimeSeries to a remote write endpoint in order
-func (prwe *PrwExporter) export(ctx context.Context, tsMap map[string]*prompb.TimeSeries) error {
-	// Calls the helper function to convert the TsMap to the desired format
-	req, err := wrapTimeSeries(tsMap)
+func (prwe *PrwExporter) export(ctx context.Context, tsMap map[string]*prompb.TimeSeries) []error {
+	var errs []error
+	// Calls the helper function to convert and batch the TsMap to the desired format
+	requests, err := wrapAndBatchTimeSeries(tsMap)
 	if err != nil {
-		return consumererror.Permanent(err)
+		errs = append(errs, consumererror.Permanent(err))
+		return errs
 	}
 
+	var wg sync.WaitGroup
+	var mtx sync.Mutex
+
+	for _, request := range requests {
```

Comment:
One thing I'm worried about this is it can create a large number of concurrent outgoing request. We should cap the max number of concurrent requests.

Reply:
Added a cap on the number of concurrent requests (with some logs) and addressed the comments you mentioned above. Here's an example of the logs produced with this [commit](https://github.com/open-o11y/opentelemetry-collector/pull/34/commits/e5a7639c41cc85754702a316fde1351c122af93d) where the cap was 3 concurrent requests.

```
total metrics to process: 194128
tsArray length: 44566
sizeOfCurrentBatch: 2999930
tsArray length: 44560
sizeOfCurrentBatch: 2999986
tsArray length: 44555
sizeOfCurrentBatch: 2999962
tsArray length: 44553
sizeOfCurrentBatch: 2999978
tsArray length: 15894
sizeOfCurrentBatch: 1069220
next batch of requests
making request 0
making request 1
making request 2
next batch of requests
making request 4
making request 3
```